graph/pull.go

```diff
@@ -149,7 +149,12 @@ func (s *TagStore) pullRepository(r *registry.Session, out io.Writer, repoInfo *
 	errors := make(chan error)
 
 	layersDownloaded := false
-	for _, image := range repoData.ImgList {
+	imgIDs := []string{}
+	defer func() {
+		s.graph.Release(imgIDs, imgIDs)
```

Comment:
we can probably add this ID as a field on the `registry.Session` type, generating a random ID on construction in `registry.NewSession()`.